LLMonFHIR/FHIR Display/OpenAIChatView.swift

```diff
@@ -87,4 +77,112 @@ struct OpenAIChatView: View {
             }
         }
     }
+    
+    private func processMultipleResourceChat() async throws {
+        let resourcesArray = await fhirStandard.resources
+        var stringResourcesArray = resourcesArray.map { "\($0.displayName) in \($0.resourceType)" }
+        stringResourcesArray.append("N/A")
+        let functionCallOutputArray = try await getFunctionCallOutputArray(stringResourcesArray)
+        if !functionCallOutputArray.contains(where: { $0.contains("N/A") }) {
+            processFunctionCallOutputArray(functionCallOutputArray: functionCallOutputArray, resourcesArray: resourcesArray)
+        }
+    }
+
+    private func getFunctionCallOutputArray(_ stringResourcesArray: [String]) async throws -> [String] {
+        let openAI = OpenAI(apiToken: openAPIComponent.apiToken ?? "")
+        
+        let functions = [
+            ChatFunctionDeclaration(
+                name: "get_resource_titles",
+                description: String(localized: "FUNCTION_DESCRIPTION"),
+                parameters: JSONSchema(
+                    type: .object,
+                    properties: [
+                        "resources": .init(type: .string, description: String(localized: "PARAMETER_DESCRIPTION"), enumValues: stringResourcesArray)
+                    ],
+                    required: ["resources"]
+                )
+            )
+        ]
+
+        let chat = [
+            Chat(role: .user, content: chat.last?.content ?? ""),
+            Chat(role: .system, content: String(localized: "FUNCTION_CONTEXT") + stringResourcesArray.rawValue)
+        ]
```

Comment:
We currently only pass in the last message here, we should pass in the complete chat here, e.g. just use the chat property to query the API.
if you want to ensure that the system message is in place, it might make more sense to add it to the chat passed in in the initializer.